contracts/icp/context-config/src/mutate.rs

```diff
@@ -169,22 +163,26 @@ fn update_application(
 fn add_members(
     signer_id: &SignerId,
     context_id: &ContextId,
+    nonce: u64,
     members: Vec<ICRepr<ContextIdentity>>,
 ) -> Result<(), String> {
     with_state_mut(|configs| {
-        // Get the context or return error if it doesn't exist
         let context = configs
             .contexts
             .get_mut(context_id)
             .ok_or_else(|| "context does not exist".to_string())?;
 
-        // Get mutable access to the members through the Guard
+        // Check nonce
+        check_and_increment_nonce(context, nonce, signer_id)?;
+
+        // Rest of the function...
         let guard_ref = context.members.get(signer_id).map_err(|e| e.to_string())?;
         let mut ctx_members = guard_ref.get_mut();
 
-        // Add each member
         for member in members {
-            ctx_members.insert(member);
+            if !ctx_members.contains_key(&member) {
+                ctx_members.insert(member, 0); // Only insert if member doesn't exist
```

Comment:
`contains_key` will do a lookup, and `insert` will do another (to return the old value which doesn't exist)

`entry(..).or_default()` works too, and avoids that situation.